stable_baselines3/common/vec_env/vec_monitor.py

```diff
@@ -0,0 +1,74 @@
+import time
+from typing import Optional, Tuple
+
+import numpy as np
+
+from stable_baselines3.common.vec_env.base_vec_env import VecEnv, VecEnvObs, VecEnvStepReturn, VecEnvWrapper
+
+
+class VecMonitor(VecEnvWrapper):
+    """
+    A vectorized monitor wrapper for *vectorized* Gym environments,
+    it is used to record the episode reward, length, time and other data.
+
+    Some environments like `openai/procgen <https://github.com/openai/procgen>`_
+    or `gym3 <https://github.com/openai/gym3>` directly initialize the
+    vectorized environments, without giving us a chance to use the `Monitor`
+    wrapper. So this class simply does the job of the `Monitor` wrapper on
+    a vectorized level.
+
+    :param venv: The vectorized environment
+    :param filename: the location to save a log file, can be None for no log
+    :param info_keywords: extra information to log, from the information return of env.step()
+    """
+
+    def __init__(
+        self,
+        venv: VecEnv,
+        filename: Optional[str] = None,
+        info_keywords: Tuple[str, ...] = (),
+    ):
+        # Avoid circular import
+        from stable_baselines3.common.monitor import ResultsWriter
+
+        VecEnvWrapper.__init__(self, venv)
+        self.episode_returns = None
+        self.episode_lengths = None
+        self.episode_count = 0
+        self.t_start = time.time()
+        if filename:
+            self.results_writer = ResultsWriter(filename, header={"t_start": self.t_start}, extra_keys=info_keywords)
```

Comment:
no env id that time ?

Reply:
![image](https://user-images.githubusercontent.com/5555347/110230263-b8e7c000-7edd-11eb-9053-dbf7d49583ce.png)

So gym3 style envs don't necessarily have the field.

Reply:
I see, but you can still do a check and use it if present, to be consistent with `Monitor`